proof_json.go

```diff
@@ -183,32 +184,46 @@ func (vp *VerkleProof) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// TODO use gencodec if that works
 type stemStateDiffMarshaller struct {
-	Stem        string           `json:"stem"`
-	SuffixDiffs SuffixStateDiffs `json:"suffixDiffs"`
+	Stem     string   `json:"stem"`
+	Suffixes string   `json:"suffixes"`
+	Current  []string `json:"current"`
+	New      []string `json:"new"`
 }
 
 func (ssd StemStateDiff) MarshalJSON() ([]byte, error) {
+	suffixes := make([]byte, 0, len(ssd.Updates)+len(ssd.Inserts))
+	for i := range ssd.Updates {
+		suffixes = append(suffixes, ssd.Updates[i].Suffix)
+	}
+	for i := range ssd.Inserts {
+		suffixes = append(suffixes, ssd.Inserts[i].Suffix)
+	}
+	sort.Slice(suffixes, func(i, j int) bool { return suffixes[i] < suffixes[j] })
```

Comment:
I think in [Type 2 there might be a bug](https://github.com/ethereum/go-verkle/pull/443/files#diff-bd86229874add74bf1b46ffcac89ca1157aa411a771321037ba2c63f84989214R197) since it's only inserting Updates, but prob we need both Updates+Inserts as I did here?
I also sort them since I'd guess that's expected.

Reply:
that's possible, I'm only focusing on the SSZ encoding at the moment. We'll pick the best method and fix it if need be.